### PR TITLE
**ONGOING DO NOT MERGE** Feature/improve documentation

### PR DESCRIPTION
**Pull request by priesgo**

### CODE REVIEW

**Comment by priesgo** [comment]
@cjohnson12 while documenting the models I spotted a couple of things that would be good to change in the models. Can you communicate these to Illumina if you find these changes appropriate?
* `ClinicalReportRD.avdl` object `AdditionalAnalaysisPanel` is now using the object `Panel` instead of defining again `panelName` and `panelVersion`.
* `InterpretationRequestRD` and `CancerInterpretationRequest` are now using the enum `Assembly` defined in `CommonInterpreted.avdl` instead of a string

**Comment by priesgo** [comment]
@Antonior26 change to nullability of population has been rolled back

**Comment by priesgo** [comment]
@Antonior26 change to nullability of population has been rolled back

**Comment by priesgo** [comment]
@cjohnson12 another model change to report to illumina:
* Fields `interpretationRequestVersion` has been changed from `string` to `int` in IR, IG and CR for both RD and cancer.

File: schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl [hunk 1]
```diff
@@ -141,22 +146,34 @@ protocol CancerParticipants {
...
+        /**
+        Tumour morphology as defined by Snomed RT (at least one morphology definition by either ICD, Snomed CT or Snomed RT must be provided)
+        */
         union {null, string} morphologySnomedRT;
 
         /**
-        One of these topography fields will be provided
+        Tumour topography as defined by ICD (at least one topography definition by either ICD, Snomed CT or Snomed RT must be provided)
         */
         union {null, string} topographyICD;
```
Comment (cjohnson12) [inline schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl]: requires an array

File: schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl [hunk 2]
```diff
@@ -141,22 +146,34 @@ protocol CancerParticipants {
...
+        /**
+        Tumour topography as defined by Snomed CT (at least one topography definition by either ICD, Snomed CT or Snomed RT must be provided)
+        */
         union {null, string} topographySnomedCT;
```
Comment (cjohnson12) [inline schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl]: requires an array

File: schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl [hunk 3]
```diff
@@ -141,22 +146,34 @@ protocol CancerParticipants {
...
+        /**
+        Tumour topography as defined by Snomed RT (at least one topography definition by either ICD, Snomed CT or Snomed RT must be provided)
+        */
         union {null, string} topographySnomedRT;
```
Comment (cjohnson12) [inline schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl]: requires an array

File: schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl [hunk 4]
```diff
@@ -141,22 +146,34 @@ protocol CancerParticipants {
...
 
         /**
-        One of these morphology fields will be provided
+        Tumour morphology as defined by ICD (at least one morphology definition by either ICD, Snomed CT or Snomed RT must be provided)
         */
         union {null, string} morphologyICD;
+        /**
+        Tumour morphology as defined by Snomed CT (at least one morphology definition by either ICD, Snomed CT or Snomed RT must be provided)
+        */
         union {null, string} morphologySnomedCT;
```
Comment (cjohnson12) [inline schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl]: switch to array

File: schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl [hunk 5]
```diff
@@ -141,22 +146,34 @@ protocol CancerParticipants {
...
+        /**
+        Tumour morphology as defined by Snomed RT (at least one morphology definition by either ICD, Snomed CT or Snomed RT must be provided)
+        */
         union {null, string} morphologySnomedRT;
```
Comment (cjohnson12) [inline schemas/IDLs/org.gel.models.participant.avro/1.1.0-SNAPSHOT/CancerParticipant.avdl]: switch to array

File: schemas/IDLs/org.gel.models.coverage.avro/0.1.0-SNAPSHOT/Coverage.avdl [hunk 6]
```diff
@@ -66,15 +86,16 @@ protocol CoverageProtocol {
...
         */
         int s;
         /**
-        Gap end position
+        Gap end position. Exclusive, 0-based
         */
         int e;
         /**
         Gap length (this information is redundant as can be calculated from start and end)
+        DEPRECATED
```
[the remark this answers is not in the document]
Comment (priesgo) [reply schemas/IDLs/org.gel.models.coverage.avro/0.1.0-SNAPSHOT/Coverage.avdl]: Following Java style of deprecation, this is a way of indicating that this field will disappear in the future. We need to this, so the model is compliant with current coverage running as of now.

File: schemas/IDLs/org.gel.models.coverage.avro/0.1.0-SNAPSHOT/Coverage.avdl [hunk 7]
```diff
@@ -88,25 +109,28 @@ protocol CoverageProtocol {
...
         */
         int s;
         /**
-        The exon padded start
+        The exon padded start. Inclusive, 0-based
         (i.e.: for a gene in the positive strand and a padding of 15bp, this will be the exon start positions minus 15 bp)
+        DEPRECATED
         */
-        int padded_s;
+        union{null, int} padded_s;
```
[the remark this answers is not in the document]
Comment (priesgo) [reply schemas/IDLs/org.gel.models.coverage.avro/0.1.0-SNAPSHOT/Coverage.avdl]: idem as above

File: schemas/IDLs/org.gel.models.coverage.avro/0.1.0-SNAPSHOT/Coverage.avdl [hunk 8]
```diff
@@ -356,6 +392,10 @@ protocol CoverageProtocol {
         */
         int gap_coverage_threshold;
         /**
+        The length of contiguous positions with low coverage to consider a gap
+        */
+        int gap_length_threshold;
```
[the remark this answers is not in the document]
Comment (priesgo) [reply schemas/IDLs/org.gel.models.coverage.avro/0.1.0-SNAPSHOT/Coverage.avdl]: The maximum integer representable with 32 bits is 9223372036854775807. There is no chromosome of that length so :)

**Pull request merged**